x-pack/plugins/code/public/sagas/structure.ts

```diff
@@ -18,15 +24,33 @@ function requestStructure(uri?: string) {
   });
 }
 
-function* handleLoadStructure(action: Action<string>) {
+function* beginPollingSymbols(action: Action<string>) {
   try {
-    const data = yield call(requestStructure, `git:/${action.payload}`);
-    yield put(loadStructureSuccess({ path: action.payload!, data }));
+    const pollingTaskId = yield fork(pollingSaga, action);
+    yield take([String(loadStructureSuccess), String(loadStructureFailed)]);
+    yield cancel(pollingTaskId);
```

Comment:
you can use cancel here, but also you can just `break` from the `while` loop below.

Reply:
I think `cancel` is better for making all the break cases together.